plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/inspections/MoveVariableDeclarationIntoWhenInspection.kt

```diff
@@ -38,18 +41,33 @@ class MoveVariableDeclarationIntoWhenInspection : AbstractKotlinInspection(), Cl
             if (action == NOTHING) return
             if (action == MOVE && !property.isOneLiner()) return
 
-            val highlightType = if (action == INLINE) INFORMATION else GENERIC_ERROR_OR_WARNING
             holder.registerProblemWithoutOfflineInformation(
                 property,
                 action.description,
                 isOnTheFly,
-                highlightType,
+                highlightType(action, expression, property),
                 TextRange.from(identifier.startOffsetInParent, identifier.textLength),
                 action.createFix(subjectExpression.createSmartPointer())
             )
         })
 }
 
+private fun highlightType(action: Action, whenExpression: KtWhenExpression, property: KtProperty): ProblemHighlightType = when (action) {
+    INLINE -> INFORMATION
+    MOVE -> {
+        val lineStartOffset = whenExpression.containingKtFile.getLineStartOffset(whenExpression.getLineNumber(), skipWhitespace = false)
+        if (lineStartOffset != null) {
+            val newWhenLength = (whenExpression.startOffset - lineStartOffset) + "when (".length + property.text.length + ") {".length
+            val rightMargin = CodeStyle.getSettings(whenExpression.project).RIGHT_MARGIN
```

Comment:
Javadoc says not to use this field directly, so I changed the usage accordingly.